src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/CreateReferralHelper.java

```diff
@@ -0,0 +1,162 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.helpers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.ecm.common.idam.models.UserDetails;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.HearingTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.ReferralTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.types.ReferralType;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.UserService;
+import uk.gov.hmcts.ethos.replacement.docmosis.utils.IntWrapper;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class CreateReferralHelper {
+    private final UserService userService;
+    private static final String GUIDANCE_DOC_LINK = "<hr>To help you complete this form, open the "
+        + "<a href=\"url\">referral guidance documents</a>";
+    private static final String HEARING_DETAILS = "<hr><h3>Hearing details %s</h3>"
+        + "<pre>Date &nbsp;&#09&#09&#09&#09&#09&nbsp; %s"
+        + "<br><br>Hearing &#09&#09&#09&#09&nbsp; %s"
+        + "<br><br>Type &nbsp;&nbsp;&#09&#09&#09&#09&#09 %s</pre>";
+
+    @Autowired
+    public CreateReferralHelper(UserService userService) {
+        this.userService = userService;
+    }
+
+    /**
+     * Formats the hearing details into HTML for ExUI to display. It's expected that there are at least one hearing
+     * already created before this event is started. Hearing details should contain the hearing date, hearing
+     * type and the track type for each hearing.
+     */
+    public void populateHearingDetails(CaseData caseData) {
+        if (CollectionUtils.isNotEmpty(caseData.getHearingCollection())) {
+            String trackType = caseData.getTrackType();
+            StringBuilder hearingDetails = new StringBuilder();
+            hearingDetails.append(GUIDANCE_DOC_LINK);
+            IntWrapper count = new IntWrapper(0);

```

Comment:
Why do we have a custom class to do the same thing as an AtomicInteger? Instead of `count.incrementAndReturnValue()` on line 55, we can use `count.incrementAndGet()`

EDIT: This function could be broke up, its very cognitively complex